### PR TITLE
rename envar, set default value

### PR DESCRIPTION
**Pull request by sandlerr**
@zendesk/runway

### CODE REVIEW

**Comment by sandlerr** [comment]
still reading the old envar - will deploy this, change the variable we pass in and then make another PR to remove the fallback.

File: .env.example [hunk 1]
```diff
@@ -23,7 +23,7 @@ GITHUB_TOKEN={fill me in}
 # BYPASS_DETAILS={optional 'Some text explaining bypass procedure'}
 # BUDDY_CHECK_FEATURE={optional, set to 1 to enable buddy_check feature; 0 to disable}
 
-# DEPLOY_MAX_MINUTES_PENDING={optional, set to 20 for max minutes a deploy is pending}
+# BUDDY_CHECK_TIME_LIMIT={optional, set to 20 for max minutes a deploy is pending}
```
Comment (grosser) [inline .env.example]: I was thinking it's a 'timeout', after a certain time it does not work anymore.
`BUDDY_CHECK_TIMEOUT={optional, default 20, how many minutes a commit can be redeployed without needing another buddy check}`

Comment (grosser) [inline .env.example]: maybe change to 30 or 15 ... 20 seems arbitrary ...

File: lib/policies/buddy_check.rb [hunk 2]
```diff
@@ -18,7 +18,7 @@ def bypass_jira_email_address
   end
 
   def deploy_max_minutes_pending
-    ENV["DEPLOY_MAX_MINUTES_PENDING"].to_i
+    ENV.fetch("BUDDY_CHECK_TIME_LIMIT", ENV.fetch("DEPLOY_MAX_MINUTES_PENDING", 20)).to_i
```
Comment (grosser) [inline lib/policies/buddy_check.rb]: Like the fallback :)

Comment (grosser) [inline lib/policies/buddy_check.rb]: maybe rename the method too

File: .env.example (hunk 1 above)
Comment (sandlerr) [inline .env.example]: so there are two different scenarios:
1. I deploy production, but within BUDDY_CHECK_TIMEOUT nobody approves my deploy. It gets cancelled.
2. I deploy production:datacenter1 with a buddy and some time later I try to deploy the same tag to production:datacenter2. If BUDDY_CHECK_TIMEOUT2 has elapsed then I need another approval.

Do we need separate variables here?

Comment (grosser) [inline .env.example]: ahhh the other value is BUDDY_CHECK_GRACE_PERIOD ... so no need for any
changes :D

Comment (grosser) [inline .env.example]: https://github.com/zendesk/samson/pull/503

**Comment by ashmckenzie** [comment]
Heyas, am chasing up https://zendesk.atlassian.net/browse/OP-12629...  is this PR OK to go out?

**Comment by sandlerr** [comment]
Hey Ash, I'd like to go ahead with this
cc @zendesk/runway

**Comment by ashmckenzie** [comment]
LGTM :+1:

File: lib/policies/buddy_check.rb (hunk 2 above)
Comment (sandlerr) [inline lib/policies/buddy_check.rb]: done

**Pull request merged**